File: Pruebas/U4/PrincipalU4/Notas.java

public class Notas {
  private Float[] notas;
  private int cantidad;
  public Notas(int cantidad) {
	  this.cantidad = cantidad;
	  this.notas = new Float[cantidad];
	  for (int i = 0; i < cantidad; i++)
		  this.notas[i] = (float)1;
  }
  public void addNota(int i, float nota) {
	  if (i < this.cantidad && i >= 0)
		  this.notas[i] = nota;
  }
  public float promedio() {
	  float suma = 0;
	  for (int i = 0; i < this.cantidad; i++) 
		  suma += this.notas[i];
	  return suma/this.cantidad;
  }
  public String diceNotas() {
	  String s = "Notas:";
	  for (Integer i = 0; i < this.cantidad; i++)
		  s += " " + this.notas[i].toString();
	  s += ". Prom:" + Float.toString(this.promedio());
	  return (s);
  }
}

File: Pruebas/U4/PrincipalPHPU4/Notas.php
<?php

class Notas
{
    
    private  $nota;
    private int $cantidad;

    public function __construct(int $cant)
    {
        $this->cantidad = $cant;
        $this->nota= array();
    }
    
    public function addNota(int $i, float $nota){
        
        if ($i< $this->cantidad && $i >=0) {
            $this->nota[$i] = $nota;
        }
        
    }
    
    public function promedio(){
        $suma = 0;
        
        for ($i = 0; $i < $this->cantidad; $i++) {
            $suma += $this->nota[$i];
        }
        return number_format($suma/$this->cantidad,1);
    }
    
    public function diceNotas(){
        $s = "Notas: ";
        for($i=0;$i<$this->cantidad;$i++){
            $s .= " " . $this->nota[$i];
        }
        $s .= ". Prom. " . number_format(floatval($this->promedio()),1);
        return $s;
    }
}

